docs/pipelines.md

```diff
@@ -465,6 +465,141 @@ There are a lot of scenarios where `WhenExpressions` can be really useful. Some
 - Checking if the name of a CI job matches
 - Checking if an optional Workspace has been provided
 
+#### Guarding a `Task` and its dependent `Tasks`
+
+When  `when` expressions evaluate to `False`, the `Task` and its dependent `Tasks` will be skipped by default while the 
+rest of the `Pipeline` will execute. Dependencies between `Tasks` can be either ordering ([`runAfter`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-the-runafter-parameter))
+or resource (e.g. [`Results`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-results)) 
+dependencies, as further described in [configuring execution order](#configuring-the-task-execution-order). The global 
+default scope of `when` expressions is set to a `Task` and its dependent`Tasks`; `scope-when-expressions-to-task` field 
+in [`config/config-feature-flags.yaml`](install.md#customizing-the-pipelines-controller-behavior) defaults to "false".
+
+**Note:** Scoping `when` expressions to a `Task` and its dependent `Tasks` is deprecated. To guard a `Task` and its 
```

Comment:
i think there's something I'm still missing here, and apologies if I'm just being picky about the wording, but my understanding is:

* Scoping when expressions to a task and the tasks are explicitly ordered to run after it (via runAfter) is deprecated
* Even with the new behavior, if a task is skipped and a subsequent task has a resource dependency on the skipped task, the subsequent task _will still be skipped_, that is: the when expression is still scoped to the task _and its (resource dependent) dependents_

At the moment I think this is the only option until we have [TEP-0048](https://github.com/tektoncd/community/pull/240) - and even then users would need to explicitly configure defaults to use that functionality

So in summary I'm not sure if it's just that I'm expecting to see this called out more explicitly in the docs (i.e. just a doc update - or point me at something in the docs im missing), or if this has been implemented with different assumptions (and there's a difference in functionality b/w this and how i thought it would work)?

Reply:
yes, that is correct that the ordering-dependent tasks should execute successfully and the resource dependencies will need default results -- added more documentation to explain this in further detail